baru_wallet/src/client.rs

```diff
@@ -0,0 +1,9 @@
+use crate::Utxo;
+use anyhow::Result;
+use async_trait::async_trait;
+use elements::{Address, TxOut};
+
+#[async_trait]
+pub trait Client {
```

Comment:
We could follow the "single trait fn" pattern here and name this `GetTxouts`.

Plus, I think the `Utxo` type should be moved into this file.